iib/web/models.py

```diff
@@ -1075,6 +1099,168 @@ def get_mutable_keys(self):
+    target_index = db.relationship('Image', foreign_keys=[target_index_id], uselist=False)
+    target_index_resolved = db.relationship(
+        'Image', foreign_keys=[target_index_resolved_id], uselist=False
+    )
+
+    __mapper_args__ = {
+        'polymorphic_identity': RequestTypeMapping.__members__['merge_index_image'].value,
+    }
+
+    @classmethod
+    def from_json(cls, kwargs, batch=None):
+        """
+        Handle JSON requests for the merge-index-image API endpoint.
+
+        :param dict kwargs: the JSON payload of the request.
+        :param Batch batch: the batch to specify with the request.
+        """
+        request_kwargs = deepcopy(kwargs)
+
+        deprecation_list = request_kwargs.pop('deprecation_list', [])
+        if not isinstance(deprecation_list, list) or any(
+            not item or not isinstance(item, str) for item in deprecation_list
+        ):
+            raise ValidationError(
+                'The "deprecation_list" value should be an empty array or an array of strings'
+            )
+
+        request_kwargs['deprecation_list'] = [
+            Image.get_or_create(pull_specification=item) for item in deprecation_list
+        ]
+
+        source_from_index = request_kwargs.pop('source_from_index', None)
+        if not (isinstance(source_from_index, str) and source_from_index):
+            raise ValidationError('"source_from_index" must be a string')
+        request_kwargs['source_from_index'] = Image.get_or_create(
+            pull_specification=source_from_index
+        )
+
+        target_index = request_kwargs.pop('target_index', None)
+        if target_index:
+            if not isinstance(target_index, str):
+                raise ValidationError('"target_index" must be a string')
+            request_kwargs['target_index'] = Image.get_or_create(pull_specification=target_index)
+
+        # Verify that `overwrite_target_index` is the correct type
+        overwrite = request_kwargs.pop('overwrite_target_index', False)
+        if not isinstance(overwrite, bool):
+            raise ValidationError('The "overwrite_target_index" value must be a boolean')
+
+        # Verify that `overwrite_target_index_token` is the correct type
+        overwrite_token = request_kwargs.pop('overwrite_target_index_token', None)
+        if overwrite_token:
+            if not isinstance(overwrite_token, str):
+                raise ValidationError('The "overwrite_target_index_token" value must be a string')
+            if overwrite_token and not overwrite:
+                raise ValidationError(
+                    'The "overwrite_target_index" value is required when'
+                    ' the "overwrite_target_index_token" value is used'
+                )
+
+        # Verify the user is authorized to use overwrite_target_index
+        # current_user.is_authenticated is only ever False when auth is disabled
+        if current_user.is_authenticated:
```

Comment:
This needs to raise an exception if the `overwrite_target_index` was supplied without the token due to you addressing https://github.com/release-engineering/iib/pull/162#discussion_r496666071.

Reply:
Done